packages/big-design/src/components/ButtonGroup/ButtonGroup.tsx

```diff
@@ -0,0 +1,153 @@
+import { MoreHorizIcon } from '@bigcommerce/big-design-icons';
+import React, { createRef, HTMLAttributes, memo, useCallback, useEffect, useMemo, useState } from 'react';
+
+import { useWindowResizeListener } from '../../hooks';
+import { MarginProps } from '../../mixins';
+import { ButtonProps } from '../Button';
+import { Dropdown } from '../Dropdown';
+import { Flex } from '../Flex';
+
+import { StyledButton, StyledFlexItem } from './styled';
+
+export interface ButtonGroupAction extends Omit<ButtonProps, 'children' | 'iconOnly' | 'iconRight' | 'iconLeft'> {
+  text: string;
+  icon?: React.ReactElement;
+}
+
+export interface ButtonGroupProps extends HTMLAttributes<HTMLDivElement>, MarginProps {
+  actions: ButtonGroupAction[];
+}
+
+const excludeIconProps = <T extends any>({
+  iconOnly,
+  iconRight,
+  iconLeft,
+  ...actionProps
+}: T): Pick<T, Exclude<keyof T, 'iconOnly' | 'iconRight' | 'iconLeft'>> => actionProps;
```

Comment:
Let's not complicate this with generics:
```suggestion
const excludeIconProps = ({
  iconOnly,
  iconRight,
  iconLeft,
  ...actionProps
}: ButtonProps & { text: string }): ButtonGroupAction => actionProps;
```

I don't really like the ` & { text: string }` part 😅 , but I guess it's okay since it's an internal implementation 🤷

Reply:
Thank you! I lil bit changed your suggestion to: `{ text: string }` => `Pick<ButtonGroupAction, 'text'>`